book/src/DEVELOPER_README.md

```diff
@@ -1,5 +1,90 @@
 # Getting Started (for Developers)
 
+## Development Principles
+
+As a piece of software that stores the identities of people, the project becomes bound to social and
+political matters. The decisions we make have consequences on many people - many who never have the
+chance to choose what software is used to store their identities (think employees in a business, or
+the users of a website).
+
+This means we have a responsibility to not only be aware of our impact on our direct users
+(developers, system administrators, dev ops, security and more) but also the impact on indirect
+consumers - many of who are unlikely to be in a position to contact us to ask for changes and help.
+
+### Ethics / Rights
+
+If you have not already, please see our documentation on [rights and ethics]
+
+[rights and ethics]: https://github.com/kanidm/kanidm/blob/master/project_docs/ethics/README.md
+
+### Humans First
+
+We must at all times make decisions that put humans first. We must respect all cultures, languages,
+and identities and how they are represented.
+
+This may mean we make technical choices that are difficult or more complex, or different to "how
+things have always been done". But we do this to ensure that all people can have their identities
+stored how they choose.
+
+For example, any user may change their name, display name and legal name at any time. Many
+applications will break as they primary key from name when this occurs. But this is the fault of the
+application. Name changes must be allowed. Our job as technical experts is to allow that to happen.
+
+We will never put a burden on the user to correct for poor designs on our part. For example, locking
+an account if it logs in from a different country unless the user logs in before hand to indicate
+where they are going. This makes the user responsible for a burden (changing the allowed login
+country) when the real problem is preventing bruteforce attacks - which can be technically solved in
```

Comment:
Weeeeelllllll there's valid reasons for blocking countries from logging in at all, for regulatory reasons, but that's slightly to the side of this description.